lib/s3_util.rb

```diff
@@ -41,7 +41,7 @@ def build_zip
   end
 
   def add_pages(zip_file, pages_node)
-    @translation.resource.pages.each do |page|
+    @translation.resource.pages.order(position: :asc).each do |page|
```

Comment:
Since you allow nulls, you might want to get explicit here about whether this sorts nulls first or last, and/or test for what you expect to happen:
http://stackoverflow.com/questions/5826210/rails-order-with-nulls-last